### PR TITLE
template modified but no access to variable networkID

### PR DESCRIPTION
**Pull request by zhongh**
The problem is that this template has no access to the networkId variable 
                and I can't find it anywhere or figure out how to resolve this.
                So it is always showing there is no network ID.

### CODE REVIEW

**Comment by zednis** [comment]
I merged the master branch into this feature branch so you will have the up-to-date 3rd-tier build.

This feature branch now has the DCO-customized edu.cornell.mannlib.vitro.webapp.controller.individual.IndividualResponseBuilder class under /src

**Comment by zhongh** [comment]
where can i find the corresponding codes about how we did it for the dcoid? I can't find a clue yet after reading related codes

**Comment by zednis** [comment]
@zhonghao1014 were you looking at [IndividualResponseBuilder.java](https://github.com/tetherless-world/dco-vivo/blob/haoAddNetworkIDUnderDCOID/src/edu/cornell/mannlib/vitro/webapp/controller/individual/IndividualResponseBuilder.java)?

**Comment by zhongh** [comment]
Yes I am...

**Comment by zednis** [comment]
@zhonghao1014 Do you have any specific questions with the code?  You can find our dco customizations in the file by searching the file for "dcoId".

**Comment by zhongh** [comment]
Thanks! My IndividualResponseBuilder.java file with some modifications for dcoId locates in Vitro/webapp/src/... I think i need to download the branching of the newest version first.

**Comment by zednis** [comment]
Do a `git pull` on this branch to pickup the current 3-tier configuration.  You should then have our customized version of the class under /src.  Do not make changes to any classes in the vivo or vitro submodules.

**Comment by mrpatrickwest** [comment]
Just FYI, in the future when you do a pull request make sure to give it a meaningful name. In this pull request could have made it "Display networkId at top of person's profile". And then your first comment could have been "The problem is that this template ..."

**Comment by zhongh** [comment]
Got it, Patrick & Stephan!

**Comment by zhongh** [comment]
So I have been looking at IndividualResponseBuilder.java. In my understanding, this is used by IndividualController.java to construct individual reponse informations. (not sure if i am right).

So now I try to edit the assembleResponse() in it. In identical manner I added these but it never works.. guidance needed!
// Trial for networkId
        Individual networkId = individual.getRelatedIndividual("http://info.deepcarbon.net/schema#networkId");
        if (networkId != null){
            body.put("networkId", networkId.getRdfsLabel());
            System.out.println("world!!!");
        }

and not sure about what     private static String predicateDcoID = dcoOntoNamespace+"dcoId" means..

**Comment by zednis** [comment]
The code:

```
private static String predicateDcoID = dcoOntoNamespace+"dcoId"
```

assigns the string value "http://info.deepcarbon.net/schema#dcoId" to variable predicateDcoID.

**Comment by zednis** [comment]
@zhonghao1014 What problem specifically are you having?

Are you getting compile errors?  Does your code modifications compile but your locally deployed VIVO is throwing errors?  Does it compile and run but the updates are having no visible affect?

**Comment by zhongh** [comment]
it compiles and runs but the updates are having no visible affect.

**Comment by zednis** [comment]
Is that an answer to my question or are you asking for clarification on what I asked?

**Comment by zhongh** [comment]
sorry for the question mark.. yes it compiles and runs but has no visible effect

**Comment by zednis** [comment]
Have you checked the vivo log file?

--Stephan

**Comment by zhongh** [comment]
which one is it?

**Comment by zhongh** [comment]
do you mean this: https://wiki.duraspace.org/display/VIVO/The+VIVO+log+file

**Comment by mrpatrickwest** [comment]
/usr/local/etc/tomcat/logs/vivo.all.log

I visited my profile page and there was nothing logged to the log file.

You need to learn how to write to the logfile and log what's going on in the code you've updated.

File: src/edu/cornell/mannlib/vitro/webapp/controller/individual/IndividualResponseBuilder.java [hunk 1]
```diff
@@ -86,7 +86,15 @@ ResponseValues assembleResponse() throws TemplateModelException {
 		Individual dcoId = individual.getRelatedIndividual("http://info.deepcarbon.net/schema#hasDcoId");
 		if (dcoId != null) {	
 			body.put("dcoId", dcoId.getRdfsLabel());
+			System.out.println("world!!!");
 		}
+		// Trial for networkId
+		Individual networkId = individual.getRelatedIndividual("http://info.deepcarbon.net/schema#networkId");
```
Comment (mrpatrickwest) [inline src/edu/cornell/mannlib/vitro/webapp/controller/individual/IndividualResponseBuilder.java]: This is NOT the URI for the networkId

http://vivo.mydomain.edu/ns#networkId

File: themes/dco/templates/individual--foaf-person.ftl [hunk 2]
```diff
@@ -91,6 +91,18 @@
                     DCO ID <a href="http://dx.deepcarbon.net/${dcoId!}" target="_blank" title="DCO-ID">${dcoId!}</a>
                 </h2>
             </#if>
+            <!-- Here below. The problem is that this template has no access to the networkId variable 
+            	and I can't find it anywhere or figure out how to resolve this.
+            	So it is always showing there is no network ID -->
+            <#if networkId??>
+            	<h2 id="networkId">
+	     	    	Network ID <a href="http://vitro.mydomain.edu/${networkId!}" target="_blank" title="NETWORK-ID"> (exists) ${networkId!} </a>
```
Comment (mrpatrickwest) [inline themes/dco/templates/individual--foaf-person.ftl]: We don't want a link to the networkId, just display the networkId. So remove the anchor tag.

File: themes/dco/templates/individual--foaf-person.ftl [hunk 3]
```diff
@@ -91,6 +91,18 @@
...
+				</h2>
+            <#else>
+			   	<h2 id="networkId">
+				    Network ID <a href="" target="_blank" title="NETWORK-ID"> N/A </a>
+				</h2>
```
Comment (mrpatrickwest) [inline themes/dco/templates/individual--foaf-person.ftl]: Reminder that we need to remove the <#else> here. If the networkId doesn't exist then don't display anything.

File: themes/dco/templates/individual--foaf-person.ftl [hunk 4]
```diff
@@ -91,6 +91,18 @@
                     DCO ID <a href="http://dx.deepcarbon.net/${dcoId!}" target="_blank" title="DCO-ID">${dcoId!}</a>
                 </h2>
             </#if>
+            <!-- Here below. The problem is that this template has no access to the networkId variable 
+            	and I can't find it anywhere or figure out how to resolve this.
+            	So it is always showing there is no network ID -->
```
Comment (mrpatrickwest) [inline themes/dco/templates/individual--foaf-person.ftl]: And remember to change the comments

File: themes/dco/templates/individual--foaf-person.ftl (hunk 3 above)
Comment (zhongh) [inline themes/dco/templates/individual--foaf-person.ftl]: yes. the #else case is just for me to know if we got access to networkId property, will be removed eventually.

File: src/edu/cornell/mannlib/vitro/webapp/controller/individual/IndividualResponseBuilder.java (hunk 1 above)
Comment (zhongh) [inline src/edu/cornell/mannlib/vitro/webapp/controller/individual/IndividualResponseBuilder.java]: yes i just realized and changed it. and exactly here i have a question, it seems we should be using 
getObjectPropertyStatements() other than getRelatedIndividual()

Comment (mrpatrickwest) [inline src/edu/cornell/mannlib/vitro/webapp/controller/individual/IndividualResponseBuilder.java]: Is networkId an object property? I don't think it is. So not sure getObjectPropertyStatements would be the thing to use here.

**Comment by zhongh** [comment]
networkId is basically something similar to, such as, skype id

**Comment by mrpatrickwest** [comment]
Why was this closed?

**Comment by zednis** [comment]
@zhonghao1014  WHY WAS THIS CLOSED?

**Pull request closed without merging**